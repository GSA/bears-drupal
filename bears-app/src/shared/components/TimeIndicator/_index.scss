.time-indicator {
  font-weight: 700;
  font-size: 16px;
  line-height: 18.8px;
  text-align: center;
  margin-bottom: 32px;
}
