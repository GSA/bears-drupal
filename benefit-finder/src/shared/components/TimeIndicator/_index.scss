@use '../../styles/colors/_index.scss' as color;

#benefit-finder {
  .time-indicator {
    color: color.$medium-grey;
    font-weight: 400;
    font-size: 16px;
    line-height: 18.8px;
    text-align: center;
    margin-bottom: 32px;
  }
}
