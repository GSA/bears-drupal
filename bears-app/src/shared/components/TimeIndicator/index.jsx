import PropTypes from 'prop-types'
import './_index.scss'

/**
 * a functional component that renders a div and expects a string description of time
 * @component
 * @param {string} timeEstimate - The inherited value for time description
 * @return {html} returns a div
 */
const TimeIndicator = ({ timeEstimate }) => {
  return <div className="time-indicator">Time to complete: {timeEstimate}</div>
}

TimeIndicator.propTypes = {
  timeEstimate: PropTypes.string,
}

export default TimeIndicator
