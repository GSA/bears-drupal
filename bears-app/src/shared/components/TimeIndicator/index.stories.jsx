import TimeIndicator from './index.jsx'

export default {
  component: TimeIndicator,
  tags: ['autodocs'],
  args: {
    timeEstimate: '10min',
  },
}

export const Primary = {}
