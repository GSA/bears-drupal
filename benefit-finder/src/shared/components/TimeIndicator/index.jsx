import PropTypes from 'prop-types'
import './_index.scss'

/**
 * a functional component that renders a div and expects a string description of time
 * @component
 * @param {string} description - translated ui values to describe time estimate component
 * @param {string} timeEstimate - The inherited value for time description
 * @return {html} returns a div
 */
const TimeIndicator = ({ timeEstimate, description }) => {
  return (
    <div className="time-indicator">
      {description} {timeEstimate}
    </div>
  )
}

TimeIndicator.propTypes = {
  description: PropTypes.string,
  timeEstimate: PropTypes.string,
}

export default TimeIndicator
