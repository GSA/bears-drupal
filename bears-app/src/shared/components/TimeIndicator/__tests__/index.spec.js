import renderer from 'react-test-renderer'
import TimeIndicator from '../index.jsx'

describe('TimeIndicator', () => {
  it('renders a match to the previous snapshot', () => {
    const component = renderer.create(<TimeIndicator />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
