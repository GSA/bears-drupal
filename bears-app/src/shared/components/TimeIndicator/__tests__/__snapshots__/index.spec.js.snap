// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeIndicator renders a match to the previous snapshot 1`] = `
<div
  className="time-indicator"
>
   
</div>
`;
